### PR TITLE
refactor: replace char_u with char 13: remove `STRLEN` part 3

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Comment by dundargoc** [comment]
Followup to https://github.com/neovim/neovim/pull/20140

File: src/nvim/diff.c [hunk 1]
```diff
@@ -1197,11 +1197,11 @@ void ex_diffpatch(exarg_T *eap)
   // Get the absolute path of the patchfile, changing directory below.
   fullname = FullName_save(eap->arg, false);
   esc_name =
-    vim_strsave_shellescape((char_u *)(fullname != NULL ? fullname : eap->arg), true, true);
+    (char *)vim_strsave_shellescape((char_u *)(fullname != NULL ? fullname : eap->arg), true, true);
 #else
   esc_name = vim_strsave_shellescape(eap->arg, true, true);
```
Comment (zeertzjq) [inline src/nvim/diff.c:1202]: Add cast

File: src/nvim/mapping.c [hunk 2]
```diff
@@ -422,7 +422,7 @@ static int str_to_mapargs(const char_u *strargs, bool is_unmap, MapArguments *ma
   char_u lhs_to_replace[256];
   STRLCPY(lhs_to_replace, to_parse, orig_lhs_len + 1);
 
-  size_t orig_rhs_len = STRLEN(rhs_start);
+  size_t orig_rhs_len = strlen(rhs_start);
   if (!set_maparg_lhs_rhs((char *)lhs_to_replace, orig_lhs_len,
                           (char *)rhs_start, orig_rhs_len, LUA_NOREF,
```
Comment (zeertzjq) [inline src/nvim/mapping.c:427]: unnecessary cast

File: src/nvim/search.c [hunk 3]
```diff
@@ -3428,9 +3428,9 @@ void find_pattern_in_path(char_u *ptr, Direction dir, size_t len, bool whole, bo
   int old_files;
   int already_searched;
   char_u *file_line;
-  char_u *line;
-  char_u *p;
-  char_u save_char;
+  char *line;
+  char *p;
```
Comment (zeertzjq) [inline src/nvim/search.c:3432]: Did you forget #20517? You need to cast to `uint8_t` when passing a char in this string to a function.

File: src/nvim/search.c [hunk 4]
```diff
@@ -3439,8 +3439,8 @@ void find_pattern_in_path(char_u *ptr, Direction dir, size_t len, bool whole, bo
   bool did_show = false;
   bool found = false;
   int i;
-  char_u *already = NULL;
-  char_u *startp = NULL;
+  char *already = NULL;
+  char *startp = NULL;
```
Comment (zeertzjq) [inline src/nvim/search.c:3443]: ditto

**Review by zeertzjq — DISMISSED** [review]

File: src/nvim/search.c (hunk 3 above)
Comment (dundargoc) [reply src/nvim/search.c:3432]: Thanks

**Pull request merged**